### ISSUE TITLE
beacon mac address

### ISSUE THREAD

**Issue by quba88** [kontaktio/kontakt-ios-sdk#41]
It is possible to get mac address from beaconManager:didRangeBeacons: inRegion: delegate method.
Something equivalent with android [getAddress](http://kontaktio.github.io/kontakt-android-sdk/3.2.2/Javadoc/com/kontakt/sdk/android/ble/device/BeaconDevice.html#getAddress--)

**Comment by stoprocent** [comment]
Hi @quba88 
No :( CoreBluetooth is not providing mac addresses of BLE devices.

**Comment by adrianz** [comment]
Hi @quba88 

TL;DR version - Developers don't have access to MAC addresses on iOS

Full version - iBeacon format specifies the Proximity UUID, Major and Minor as only available identifiers and `CLBeacon` object provides only this as a reference for a beacon. There is no other way to detect iBeacons on iOS other than using the Core Location API, so `KTKBeaconManager` can't provide any other information that what iOS is giving us.

If a beacon is a connectable Bluetooth device (like our beacons are) it's possible to detect it via Core Bluetooth, and that's what `KTKDevicesManager` is doing when you want to e.g. configure a beacon. But in this case MAC address is not available as well – for privacy reasons iOS generates unique identifiers for Bluetooth devices your iPhone/iPad detects and don't expose MAC addresses to developers.
